soccer/src/soccer/strategy/agent/position/seeker.cpp

```diff
@@ -163,14 +163,20 @@ double Seeker::eval_point(rj_geometry::Point ball_pos, rj_geometry::Point curren
     float min_path_dist = 10000;
     for (auto bot : world_state->their_robots) {
         rj_geometry::Point opp_pos = bot.pose.position();
-        min_robot_dist = std::min(min_robot_dist, current_point.dist_to(opp_pos));
-        min_path_dist = std::min(min_path_dist, pass_path.dist_to(opp_pos));
+        auto robot_dist = current_point.dist_to(opp_pos);
+        min_robot_dist = std::min(min_robot_dist, robot_dist);
+        auto path_dist = pass_path.dist_to(opp_pos);
+        min_path_dist = std::min(min_path_dist, path_dist);
     }
 
     for (auto bot : world_state->our_robots) {
         rj_geometry::Point ally_pos = bot.pose.position();
-        min_robot_dist = std::min(min_robot_dist, current_point.dist_to(ally_pos));
-        min_path_dist = std::min(min_path_dist, pass_path.dist_to(ally_pos));
+        auto robot_dist = current_point.dist_to(ally_pos);
+        // if dist is 0, then bot must be seeker (self ) robot, so should ignore
+        if (robot_dist == 0) continue;
+        min_robot_dist = std::min(min_robot_dist, robot_dist);
+        auto path_dist = pass_path.dist_to(ally_pos);
+        min_path_dist = std::min(min_path_dist, path_dist);
     }
 
     min_path_dist = 0.1f / min_path_dist;
```

Comment:
what does this do?

Reply:
The oscillation problem occurs because the for loop above includes the current seeker robot. The dist from current point to the current robot is 0, leading to divide by 0 problems. This skips over that case.